File: src/dependencies/crazyflie_hardware/src/crtp/crtp/logic/console_logic.py
from .logic import Logic
from crtp.crtp_link import CrtpLink
from crtp.packers.console_packer import ConsolePacker
from crtp.packers.crtp_packer import CrtpPacker

from typing import Callable


class ConsoleLogic(Logic):

    def __init__(
        self, crtp_packer_factory: Callable[[int], CrtpPacker], crtp_link: CrtpLink
    ):
        super().__init__(crtp_link)
        self.packer = ConsolePacker(crtp_packer_factory)

    def send_consolepacket(self):
        packet = self.packer.consolepacket()
        self.link.send_packet_no_response(packet)

File: src/dependencies/crazyflie_hardware/src/crtp_cpp/src/crtp_cpp/logic/console_logic.cpp
#include "crtp_cpp/logic/console_logic.hpp"

ConsoleLogic::ConsoleLogic(CrtpLink *crtp_link)
    : Logic(crtp_link),
      packer(ConsolePacker())
{
    link->add_callback(PORT_CONSOLE, std::bind(&ConsoleLogic::crtp_response_callback, this, std::placeholders::_1));
}

void ConsoleLogic::send_consolepacket()
{
    CrtpRequest request;
    request.packet = packer.consolepacket();
    link->send_packet_no_response(request);
}